### PR TITLE
DOC:added footcite

### PR DESCRIPTION
**Pull request by apoorva6262**
#### Reference issue
issue #8934

#### What does this implement/fix?
added footcite 

#### Additional information
none

### CODE REVIEW

**Comment by drammock** [comment]
You'll also want to change the hard-coded biblio entry in the `References` part of the docstring (line 999 I think)

**Comment by sappelhoff** [comment]
> You'll also want to change the hard-coded biblio entry in the References part of the docstring (line 999 I think)

yes, starts at line 999

```
    .. [1] Darvas F, Ermer JJ, Mosher JC, Leahy RM (2006). "Generic head
           models for atlas-based EEG source analysis."
           Human Brain Mapping 27:129-143
```


so you probably want to look into https://github.com/mne-tools/mne-python/blob/main/doc/references.bib to see whether that reference is already there.

If yes: replace the one in `mne/transforms.py` with a footcite reference
If no: first add that reference to `references.bib` and THEN replace the one in `mne/transforms.py`

**Comment by sappelhoff** [comment]
ups I see now that my comment https://github.com/mne-tools/mne-python/pull/9073#issuecomment-799649613 was off target :) good job on not getting trapped by that :-)

File: mne/connectivity/spectral.py [hunk 1]
```diff
@@ -699,60 +699,42 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
...
 
-        'imcoh' : Imaginary coherence [1]_ given by::
+        'imcoh' : Imaginary coherence :footcite:`NolteEtAl2004`. given by::
 
                       Im(E[Sxy])
             C = ----------------------
                 sqrt(E[Sxx] * E[Syy])
 
-        'plv' : Phase-Locking Value (PLV) [2]_ given by::
+        'plv' : Phase-Locking Value (PLV) :footcite:`LachauxEtAl1999`. given by::
```
Comment (larsoner) [inline mne/connectivity/spectral.py:708]:
```suggestion
        'plv' : Phase-Locking Value (PLV) :footcite:`LachauxEtAl1999`
        given by::
```

File: mne/connectivity/spectral.py [hunk 2]
```diff
@@ -699,60 +699,48 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
...
+        'imcoh' : Imaginary coherence :footcite:`NolteEtAl2004` 
+        given by::
 
                       Im(E[Sxy])
             C = ----------------------
                 sqrt(E[Sxx] * E[Syy])
-
-        'plv' : Phase-Locking Value (PLV) [2]_ given by::
-
+
```
Comment (larsoner) [inline mne/connectivity/spectral.py:708]:
```suggestion
```

**Review by larsoner — COMMENTED** [review]
Looks like either the ref is missing, or the enyay is not handled properly:
```
/home/circleci/project/doc/generated/mne.connectivity.spectral_connectivity.rst:184: WARNING: could not find bibtex key "BruñaEtAl2018"
```

**Comment by drammock** [comment]
> Looks like either the ref is missing, or the enyay is not handled properly:
> 
> ```
> /home/circleci/project/doc/generated/mne.connectivity.spectral_connectivity.rst:184: WARNING: could not find bibtex key "BruñaEtAl2018"
> ```

indeed it looks like it's missing from `doc/references.bib`.  You'll need to add this:

```
@article{BrunaEtAl2018,
  title = {Phase Locking Value Revisited: Teaching New Tricks to an Old Dog},
  shorttitle = {Phase Locking Value Revisited},
  author = {Bru{\~n}a, Ricardo and Maest{\'u}, Fernando and Pereda, Ernesto},
  year = {2018},
  volume = {15},
  pages = {056011},
  doi = {10.1088/1741-2552/aacfe4},
  journal = {Journal of Neural Engineering},
  number = {5}
}
```

Please put it in the file roughly where it belongs alphabetically (by first author surname).  This helps avoid merge conflicts when others are adding refs.  I'm not certain if the key needs to be ASCII or not.

File: mne/connectivity/spectral.py [hunk 3]
```diff
@@ -699,60 +699,47 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
...
-
-        'plv' : Phase-Locking Value (PLV) [2]_ given by::
-
+        'plv' : Phase-Locking Value (PLV)
+        :footcite:`LachauxEtAl1999` given by::
             PLV = |E[Sxy/|Sxy|]|
 
-        'ciplv' : corrected imaginary PLV (icPLV) [3]_ given by::
+        'ciplv' : corrected imaginary PLV (icPLV)
+        :footcite:`BruñaEtAl2018` given by::
```
Comment (sappelhoff) [inline mne/connectivity/spectral.py:713]: You have added the reference to `references.bib`, but the "citation key" that you added is now `BrunaEtAl2018`, without that special `ñ`

```suggestion
        :footcite:`BrunaEtAl2018` given by::
```

File: mne/connectivity/spectral.py [hunk 4]
```diff
@@ -699,60 +699,47 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
...
+        given by::
 
                       Im(E[Sxy])
             C = ----------------------
                 sqrt(E[Sxx] * E[Syy])
-
-        'plv' : Phase-Locking Value (PLV) [2]_ given by::
-
+        'plv' : Phase-Locking Value (PLV)
+        :footcite:`LachauxEtAl1999` given by::
```
Comment (sappelhoff) [inline mne/connectivity/spectral.py:709]: you removed a blank line above here that needs to be there to distinguish the indented block (the formula) with the text that comes below

```suggestion

        'plv' : Phase-Locking Value (PLV)
        :footcite:`LachauxEtAl1999` given by::
```

Comment (britta-wstnr) [reply mne/connectivity/spectral.py:709]: @apoorva6262 you are still missing this blank line, I think. That's why CIs did not come back green. (You added one in line 710 but not 708)

File: doc/overview/migrating.rst [hunk 5]
```diff
@@ -37,6 +37,8 @@ below which and above which to filter out data.
 +---------------------+----------------------------------------------------------+--------------------------------------------------------------------------------------------------+
 | Filter data         | ``EEG = pop_eegfiltnew(EEG, l_freq, h_freq);``           | :func:`raw.filter(l_freq, h_freq) <mne.io.Raw.filter>`                                           |
 +---------------------+----------------------------------------------------------+--------------------------------------------------------------------------------------------------+
+| Filter data         | ``EEG = pop_eegfiltnew(EEG, l_freq, h_freq);``           | :func:`raw.filter(l_freq, h_freq) <mne.io.Raw.filter>`                                           |
++---------------------+----------------------------------------------------------+--------------------------------------------------------------------------------------------------+
```
Comment (sappelhoff) [inline doc/overview/migrating.rst:41]: @apoorva6262 it seems like you forgot to switch branches before working on a new pull request: This was probably meant to be part of a new project, right?

**Review by sappelhoff — CHANGES_REQUESTED** [review]
@apoorva6262 This was the commit where you added some changes that were probably meant for another PR: https://github.com/mne-tools/mne-python/pull/9073/commits/32b835e263fd474a04997d9bb620cdd709fdb1b4

I commented on this here: https://github.com/mne-tools/mne-python/pull/9073/commits/32b835e263fd474a04997d9bb620cdd709fdb1b4#r595905422

I think you should "git revert" that commit. That'd work like this from the command line:

1. make sure all current changes are committed and the working directory is clean (do `git status` for that)
2. do `git revert 32b835e263fd474a04997d9bb620cdd709fdb1b4` and commit this change --> that will "undo" your "wrong" changes
3. Your commit with "wrong" changes also included some correct changes .. you will have to "re-do" them, I list them below for your convenience:


in `doc/references.bib` and in `mne/connectivity/spectral.py` chance `BruñaEtAl2018` -> `BrunaEtAl2018`

Finally, please take a look at what Britta wrote here: https://github.com/mne-tools/mne-python/pull/9073#discussion_r594955420

After that, this PR should be good to go I think.

**Comment by drammock** [comment]
superseded by #9153

**Pull request closed without merging**